app/controllers/tracks/approaches_controller.rb

```diff
@@ -2,15 +2,12 @@ class Tracks::ApproachesController < ApplicationController
   include UseTrackExerciseSolutionConcern
   before_action :use_solution
   before_action :use_approach, only: :show
-  before_action :guard_accessible!, except: :tooltip_locked
+  before_action :guard_accessible!
```

Comment:
Is this deliberate?

Could you also move the `skip_...` above these `before_action`s pls.